### ISSUE TITLE
trajectory_aggregator.py Clustering

### ISSUE THREAD

**Issue by stevemarin** [anitagraser/movingpandas#74]
Thank you for movingpandas!  And thank you for the wonderful tutorials!

This is not an issue, just a question regarding the clustering of significant points.  What's the reason behind using the _Grid class to perform the clustering?  Would swapping in an sklearn clustering algorithms, like DBSCAN, cause issues I am not understanding?  Is this point to just avoid a dependency on sklearn and scipy?  I am not experiencing any issues, I'm just curious about the design choices to verify my own understanding.

Thanks again!

**Comment by anitagraser** [comment]
Hi Steve!

Indeed there's a reason but it's not a software engineering reason. It's explained in the paper that I based this development on:  

Andrienko, N., & Andrienko, G. (2011). Spatial generalization and aggregation of massive movement data. IEEE Transactions on visualization and computer graphics, 17(2), 205-219.

There's a shorter summary in:

https://anitagraser.com/2016/11/07/movement-data-in-gis-3-visualizing-massive-trajectory-datasets/

Basically, with DBSCAN, we cannot control the size of resulting clusters. This means that we could end up with some large clusters that take over large areas, thus making it impossible to see any patterns within those areas.

**Comment by stevemarin** [comment]
Got it!  Thank you very much for the reference!
